x-pack/plugins/security_solution/server/lib/detection_engine/schemas/rule_converters.ts

```diff
@@ -298,14 +302,8 @@ export const internalRuleToAPIResponse = (
     // Type specific security solution rule params
     ...typeSpecificCamelToSnake(rule.params),
     // Actions
-    throttle: transformFromAlertThrottle(rule),
-    actions:
-      rule?.actions.map((action) => ({
-        group: action.group,
-        id: action.id,
-        action_type_id: action.actionTypeId,
-        params: action.params,
-      })) ?? [],
+    throttle: transformFromAlertThrottle(rule, legacyRuleActions),
+    actions: transformActions(rule.actions, legacyRuleActions),
```

Comment:
NOTE: Double this logic for me. Exercising looked ok so far.

Reply:
NOTE: I added unit tests around this part within transformActions, so I don't think checking the logic is needed anymore. Resolving.